keras_nlp/models/xlnet/xlnet_encoder.py

```diff
@@ -0,0 +1,206 @@
+
+    Args:
+        num_heads: int, the number of heads in the
+            `keras.layers.TwoStreamRelativeAttention` layer.
+        hidden_dim: int, the size hidden states.
+        head_dim: int, the size of each attention head.
+        intermediate_dim: int, the hidden size of feedforward network.
+        dropout: float, defaults to 0.0 the dropout value, shared by
+            `keras.layers.TwoStreamRelativeAttention` and feedforward network.
+        activation: string or `keras.activations`, defaults to "gelu". the
+            activation function of feedforward network.
+        layer_norm_epsilon: float, defaults to 1e-12. The epsilon value in layer
+            normalization components.
+        kernel_initializer_range: int, defaults to 0.02. The kernel initializer
+            range for the dense and relative attention layers.
+        bias_initializer: string or `keras.initializers` initializer,
+            defaults to "zeros". The bias initializer for
+            the dense and multiheaded relative attention layers.
+        name: string, defaults to None. The name of the layer.
+        **kwargs: other keyword arguments.
+
+    References:
+     - [XLNet: Generalized Autoregressive Pretraining for Language Understanding](https://arxiv.org/abs/1906.08237)
+    """
+
+    def __init__(
+        self,
+        num_heads,
+        hidden_dim,
+        head_dim,
+        intermediate_dim,
+        dropout=0.0,
+        activation="gelu",
+        layer_norm_epsilon=1e-12,
+        kernel_initializer_range=0.02,
+        bias_initializer="zeros",
+        name=None,
+        **kwargs
+    ):
+        super().__init__(name=name, **kwargs)
+        self.num_heads = num_heads
+        self.hidden_dim = hidden_dim
+        self.head_dim = head_dim
+        self.intermediate_dim = intermediate_dim
+        self.dropout = dropout
+        self.activation = activation
+        self.layer_norm_epsilon = layer_norm_epsilon
+        self.kernel_initializer_range = kernel_initializer_range
+        self.bias_initializer = keras.initializers.get(bias_initializer)
+        self.kernel_initializer = xlnet_kernel_initializer(
+            self.kernel_initializer_range
+        )
+
+    def build(self, input_shape):
+        # Attention Part
+        self.relative_attention = TwoStreamRelativeAttention(
+            num_heads=self.num_heads,
+            key_dim=self.head_dim,
+            kernel_initializer=self.kernel_initializer,
+            bias_initializer=self.bias_initializer,
+            name="rel_attn",
+        )
+        self.layer_norm = keras.layers.LayerNormalization(
+            epsilon=self.layer_norm_epsilon, name="layer_norm_rel_attn"
+        )
+        self.dropout = keras.layers.Dropout(self.dropout)
+
+        # Feed-Forward Part
+        self.layer_norm_ff = keras.layers.LayerNormalization(
+            epsilon=self.layer_norm_epsilon, name="layer_norm_ff"
+        )
+        self.layer_1_ff = tf.keras.layers.Dense(
```

Comment:
Can we rename this to something compilant to our design, like `feedforward_intermediate_dense` and `feedforward_output_dense` ?